lib/Sema/CSSimplify.cpp

```diff
@@ -960,9 +960,26 @@ ConstraintSystem::TypeMatchResult constraints::matchCallArguments(
 
   for (unsigned paramIdx = 0, numParams = parameterBindings.size();
        paramIdx != numParams; ++paramIdx){
-    // Skip unfulfilled parameters. There's nothing to do for them.
-    if (parameterBindings[paramIdx].empty())
+    // If a parameter is unfulfilled, validate the default argument
+    if (parameterBindings[paramIdx].empty()) {
+      if (!isa<FuncDecl>(callee))
+        continue;
+
+      auto allParams = dyn_cast<FuncDecl>(callee)->getParameters()->getArray();
+      for (auto param : allParams) {
```

Comment:
We know exactly at what index parameter without argument is located so there is no need for a loop.